src/components/Navigation/NavigationItemTrigger/NavigationItemTrigger.tsx

```diff
@@ -51,6 +51,18 @@ const NavigationItemTrigger = forwardRef<HTMLLIElement, NavigationItemTriggerPro
     const navListElement = findChildrenOfType<NavigationListProps>(children, NavigationList);
     const { closeMenu } = React.useContext(HeaderContext);
 
+    React.useEffect(() => {
```

Comment:
@zhenpanda could you remove this `useEffect` and instead just do this on the callback functions that handle the `setSubmenuOpened`